examples/modeling_features/020-solid_model.py

```diff
@@ -0,0 +1,185 @@
+model.update()
+model.solid_mesh.to_pyvista().plot(show_edges=True)
+
+
+def create_virtual_geometry_from_file(
+    example_key: ExampleKeys,
+) -> tuple[CADGeometry, VirtualGeometry]:
+    """Create a CAD geometry and virtual geometry."""
+    geometry_file = get_example_file(example_key, WORKING_DIR)
+    geometry_obj = workflow.add_cad_geometry_from_local_file(geometry_file)
+    workflow.model.update()
+    virtual_geometry = model.create_virtual_geometry(
+        name="thickness_virtual_geometry", cad_components=geometry_obj.root_shapes.values()
+    )
+    return geometry_obj, virtual_geometry
+
+
+def plot_model_with_geometry(cad_geometry: CADGeometry, cad_geom_opacity: float = 0.1):
+    """Plot solid model and geometry."""
+    plotter = pyvista.Plotter()
+    geom_mesh = cad_geometry.visualization_mesh.to_pyvista()
+    plotter.add_mesh(geom_mesh, color="green", opacity=cad_geom_opacity)
+    edges = geom_mesh.extract_feature_edges()
+    plotter.add_mesh(edges, color="white", line_width=4)
+    plotter.add_mesh(edges, color="black", line_width=2)
+    plotter.add_mesh(workflow.model.solid_mesh.to_pyvista(), show_edges=True)
+    plotter.show()
+
+
+# %%
+# Snap the top to a geometry
+# --------------------------
+#
+# The :class:`.SnapToGeometry` allows to shape the bottom or top of the solid model.
+# First, import the geometry and then add the snap-to feature to the solid model.
+snap_to_geom, snap_to_virtual_geom = create_virtual_geometry_from_file(ExampleKeys.SNAP_TO_GEOMETRY)
+solid_model.create_snap_to_geometry(
+    name="Snap-to Geometry",
+    cad_geometry=snap_to_virtual_geom,
+    orientation_type=SnapToGeometryOrientationType.TOP,
+    oriented_selection_set=model.oriented_selection_sets["oss"],
+)
+
+model.update()
+plot_model_with_geometry(snap_to_geom, 0.5)
+
+# %%
+# Shape the walls
+# ---------------
+#
+# The :class:`.ExtrusionGuide` is used to shape the side walls of the solid model.
+# The feature can be defined by a direction as shown here or through a geometry.
+edge_set = model.create_edge_set(
+    name="Edge Set",
+    edge_set_type=EdgeSetType.BY_REFERENCE,
+    element_set=model.element_sets["All_Elements"],
+    limit_angle=30,
+    origin=(0.05, 0, 0),
+)
+solid_model.create_extrusion_guide(
+    name="Extrusion Guide",
+    edge_set=edge_set,
+    extrusion_guide_type=ExtrusionGuideType.BY_DIRECTION,
+    direction=(-0.5, 1, 0),
+    radius=0.005,
+    depth=0.6,
+)
+model.update()
+model.solid_mesh.to_pyvista().plot(show_edges=True)
+
+# %%
+# Cut-off an edge
+# ---------------
+#
+# The :class:`.CutOffGeometry` is used to crop elements from the solid model.
+cutoff_cad_geom, cutoff_virtual_geom = create_virtual_geometry_from_file(
+    ExampleKeys.CUT_OFF_GEOMETRY_SOLID_MODEL
+)
+solid_model.create_cut_off_geometry(
+    name="Cut-off Geometry",
+    cad_geometry=cutoff_virtual_geom,
+    orientation_type=CutOffGeometryOrientationType.UP,
+)
+
+model.update()
+plot_model_with_geometry(cutoff_cad_geom)
```

Comment:
Nice example!

I'm wondering if we should show how to plot on a partial mesh in the same example, e.g. plotting the thickness of a single analysis ply.

Reply:
Done.